conan/tools/cmake/layout.py

```diff
@@ -49,17 +50,11 @@ def get_build_folder_custom_vars(conanfile):
                                     default=[], check_type=list)
     ret = []
     for s in build_vars:
+        group, var = s.split(".", 1)
```

Comment:
what if it doesn't have a dot, will it raise with a weird exception here?

Reply:
Yes, but that was already happening before, nothing new, and I didn't want to add extra noise and tests, as this PR is already quite potentially breaking.